File: cgi-bin/put.php
<?php

// Get the file content from the request body
$fileContent = file_get_contents("php://stdin");

// Specify the directory where the file will be saved
$uploadDir = $_ENV['UPLOAD_DIR'];

// Specify the file name (you may want to use a different naming convention)
$fileName = $_ENV['FILENAME'];

// Specify the full path of the file
$filePath = $uploadDir . $fileName;

// Check if the file already exists
if (file_exists($filePath)) {
    // File already exists, so we are updating it
    // Overwrite the existing file with the new content
    if (file_put_contents($filePath, $fileContent) !== false) {
        // Set permissions for the uploaded file (optional)
        chmod($filePath, 0644); // Adjust permissions as needed
        exit(204); // Set HTTP response code to 204 (No Content)
    } else {
       // echo "Error: Failed to update file.\n";
        exit(0); // Internal Server Error
        
    }
} else {
    // File does not exist, so we are creating a new file
    // Save the file to the specified directory
    if (file_put_contents($filePath, $fileContent) !== false) {
        // Set permissions for the uploaded file (optional)
        chmod($filePath, 0644); // Adjust permissions as needed
        exit(201); // Set HTTP response code to 201 (Created)
        
    } else {
        exit(0); // Internal Server Error
    }
}
?>

File: cgi-bin/put.py
import os
import sys

fileContent = sys.stdin.read()
uploadDir = os.environ['UPLOAD_DIR']

fileName = os.environ['FILENAME']
filePath = uploadDir + fileName

if os.path.isfile(filePath):
    f = open(filePath, "w")
    f.write(fileContent)
    f.close()
    os.chmod(filePath, 0o664)
    print("Status: 204 No Content")

elif not os.path.isfile(filePath):
    f = open(filePath, "x")
    f.write(fileContent)
    f.close()
    os.chmod(filePath, 0o664)
    print("Status: 201 Created")
